2021-01-22 09:42:11+0100 [-] Log opened.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:119: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:120: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:121: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:122: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:123: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/ConnectWindow.py:124: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:42:12+0100 [-] /home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/__init__.py:81: wx.core.wxPyDeprecationWarning: Call to deprecated item __call__. Use :meth:`EvtHandler.Bind` instead.
2021-01-22 09:43:09+0100 [-] Starting factory <guiclient.LLawClientFactory object at 0x7f7dc83cc7f0>
2021-01-22 09:43:18+0100 [-] Stopping factory <guiclient.LLawClientFactory object at 0x7f7dc83cc7f0>
2021-01-22 09:49:58+0100 [-] Starting factory <guiclient.LLawClientFactory object at 0x7f7dc83cc9a0>
2021-01-22 09:49:58+0100 [Uninitialized] Unhandled Error
	Traceback (most recent call last):
	  File "/usr/lib/python3/dist-packages/twisted/python/log.py", line 103, in callWithLogger
	    return callWithContext({"system": lp}, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/log.py", line 86, in callWithContext
	    return context.call({ILogContext: newCtx}, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 122, in callWithContext
	    return self.currentContext().callWithContext(ctx, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 85, in callWithContext
	    return func(*args,**kw)
	--- <exception caught here> ---
	  File "/usr/lib/python3/dist-packages/twisted/internet/_threadedselect.py", line 276, in _doReadOrWrite
	    why = getattr(selectable, method)()
	  File "/usr/lib/python3/dist-packages/twisted/internet/tcp.py", line 627, in doConnect
	    self._connectDone()
	  File "/usr/lib/python3/dist-packages/twisted/internet/tcp.py", line 656, in _connectDone
	    self.protocol.makeConnection(self)
	  File "/usr/lib/python3/dist-packages/twisted/internet/protocol.py", line 514, in makeConnection
	    self.connectionMade()
	  File "/home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/Protocol.py", line 70, in connectionMade
	    self.sendTokens(self.genTag(), "protocol", PROTOCOL_VERSION)
	  File "/home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/Protocol.py", line 146, in sendTokens
	    self.sendLine(str(s))
	  File "/usr/lib/python3/dist-packages/twisted/protocols/basic.py", line 476, in sendLine
	    return self.transport.writeSequence((line, self.delimiter))
	  File "/usr/lib/python3/dist-packages/twisted/internet/_newtls.py", line 204, in writeSequence
	    FileDescriptor.writeSequence(self, iovec)
	  File "/usr/lib/python3/dist-packages/twisted/internet/abstract.py", line 377, in writeSequence
	    raise TypeError("Data must not be unicode")
	builtins.TypeError: Data must not be unicode
	
2021-01-22 09:49:58+0100 [Uninitialized] Unhandled Error
	Traceback (most recent call last):
	  File "/usr/lib/python3/dist-packages/wx/core.py", line 2167, in MainLoop
	    rv = wx.PyApp.MainLoop(self)
	  File "/usr/lib/python3/dist-packages/wx/core.py", line 3285, in <lambda>
	    lambda event: event.callable(*event.args, **event.kw) )
	  File "/usr/lib/python3/dist-packages/twisted/internet/_threadedselect.py", line 227, in _interleave
	    getattr(self, '_process_' + msg)(*args)
	  File "/usr/lib/python3/dist-packages/twisted/internet/_threadedselect.py", line 199, in _process_Notify
	    _logrun(selectable, _drdw, selectable, method, dct)
	--- <exception caught here> ---
	  File "/usr/lib/python3/dist-packages/twisted/python/log.py", line 103, in callWithLogger
	    return callWithContext({"system": lp}, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/log.py", line 86, in callWithContext
	    return context.call({ILogContext: newCtx}, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 122, in callWithContext
	    return self.currentContext().callWithContext(ctx, func, *args, **kw)
	  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 85, in callWithContext
	    return func(*args,**kw)
	  File "/usr/lib/python3/dist-packages/twisted/internet/_threadedselect.py", line 286, in _doReadOrWrite
	    self._disconnectSelectable(selectable, why, method == "doRead")
	  File "/usr/lib/python3/dist-packages/twisted/internet/posixbase.py", line 258, in _disconnectSelectable
	    selectable.connectionLost(failure.Failure(why))
	  File "/usr/lib/python3/dist-packages/twisted/internet/tcp.py", line 519, in connectionLost
	    self._commonConnection.connectionLost(self, reason)
	  File "/usr/lib/python3/dist-packages/twisted/internet/tcp.py", line 327, in connectionLost
	    protocol.connectionLost(reason)
	  File "/home/horald/prgentw/python/londonlaw3/londonlaw/guiclient/Protocol.py", line 63, in connectionLost
	    if self._state != "shutdown":
	builtins.AttributeError: 'LLawClientProtocol' object has no attribute '_state'
	
2021-01-22 09:50:16+0100 [-] App closed...
